lib/EnsEMBL/REST/Model/Variation.pm

```diff
@@ -192,7 +192,7 @@ sub get_gene_phenotype_info {
 
   foreach my $phen (@pfs) {
     my $hash = $self->phen_as_hash($phen);
-    my $key = join("", sort values %$hash);
+    my $key = join("", sort grep {$_} values %$hash);
```

Comment:
If you understand all Perl primitives, you understand all code right? No comments about circumstance required at all QED. Commenting on at least the purpose of the sub{} will save someone some brain glucose in future.